src/frontend/packages/git/src/shared/scm/scm-base.ts

```diff
@@ -21,25 +21,29 @@ export abstract class BaseSCM {
 
   constructor(public publicApiUrl: string) { }
 
+  public setPublicApi(url: string) {
+    this.publicApiUrl = url
+  }
+
   public getPublicApi(): string {
     return this.publicApiUrl;
   }
 
-  public getAPI(): Observable<GitApiRequest> {
+  public getAPI(options?: HttpOptions): Observable<GitApiRequest> {
     return this.getEndpoint(this.endpointGuid).pipe(
       map(endpoint => {
         if (!endpoint) {
           // No endpoint, use the default or overwritten public api associated with this type
           return {
             url: this.getPublicApi(),
-            requestArgs: {}
+            requestArgs: {...options}
           };
         }
         // We have an endpoint so always proxy via backend
         return {
           url: `${commonPrefix}/${endpoint.guid}`,
           requestArgs: {
-            ... new HttpOptions(),
+            ... (options || new HttpOptions()),
```

Comment:
with default param for options:
```suggestion
            ... options,
```

Reply:
Done 👍🏻